### ISSUE TITLE
`startLines` in `tag-lines` is not a replacement of `newline-after-description` when the jsdoc is a single line

### ISSUE THREAD

**Issue by cpcallen** [gajus/eslint-plugin-jsdoc#1036]
## Expected behaviour

The rule `'jsdoc/tag-lines': ['error', 'any', {'startLines': 1}]` is expected to be equivalent to the previous rule `'jsdoc/newline-after-description': ['error']`

## Actual behaviour

It generates errors for one-line JSDocs containing tags.

## ESLint sample

```js
/** A class. */
class _Foo {
  /** @param arg Argument. */
  conststructor(arg: string) {
    console.log(arg);
  }
}
```
generates error:
```
test.ts
  3:1  error  Expected 1 lines after block description  jsdoc/tag-lines
```
No error was generated with previous `newline-after-description` rule.

## ESLint Config

```js
{
  "env": {
    "es2020": true,
    "browser": true
  },
  "globals": {
    "goog": true,
    "exports": true
  },
  "parser": "<redacted>/node_modules/@typescript-eslint/parser/dist/index.js",
  "parserOptions": {
    "project": "./tsconfig.json",
    "tsconfigRootDir": ".",
    "ecmaVersion": 2020,
    "sourceType": "module"
  },
  "plugins": [
    "jsdoc",
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/no-namespace": [
      "off"
    ],
    "no-invalid-this": [
      "off"
    ],
    "@typescript-eslint/no-invalid-this": [
      "error"
    ],
    "@typescript-eslint/no-non-null-assertion": [
      "off"
    ],
    "no-unused-vars": [
      "off",
      {
        "args": "after-used",
        "varsIgnorePattern": "^_",
        "argsIgnorePattern": "^_"
      }
    ],
    "@typescript-eslint/no-unused-vars": [
      "warn",
      {
        "argsIgnorePattern": "^_",
        "varsIgnorePattern": "^_"
      }
    ],
    "func-call-spacing": [
      "off",
      "never"
    ],
    "@typescript-eslint/func-call-spacing": [
      "warn"
    ],
    "@typescript-eslint/no-explicit-any": [
      "off"
    ],
    "require-jsdoc": [
      "off",
      {
        "require": {
          "FunctionDeclaration": true,
          "MethodDefinition": true,
          "ClassDeclaration": true,
          "ArrowFunctionExpression": false,
          "FunctionExpression": false
        }
      }
    ],
    "@typescript-eslint/ban-types": [
      "off"
    ],
    "@typescript-eslint/no-empty-function": [
      "off"
    ],
    "@typescript-eslint/no-empty-interface": [
      "off"
    ],
    "valid-jsdoc": [
      "off",
      {
        "requireParamDescription": false,
        "requireReturnDescription": false,
        "requireReturn": false,
        "prefer": {
          "returns": "return"
        },
        "requireReturnType": true,
        "requireParamType": true
      }
    ],
    "jsdoc/require-param-type": [
      "off"
    ],
    "jsdoc/require-returns-type": [
      "off"
    ],
    "jsdoc/require-param-description": [
      "off"
    ],
    "jsdoc/require-returns": [
      "off"
    ],
    "jsdoc/require-param": [
      "off"
    ],
    "jsdoc/require-jsdoc": [
      "warn",
      {
        "enableFixer": false,
        "publicOnly": true,
        "checkConstructors": true,
        "checkGetters": true,
        "checkSetters": true,
        "exemptEmptyConstructors": false,
        "exemptEmptyFunctions": false,
        "fixerMessage": "",
        "require": {
          "ArrowFunctionExpression": false,
          "ClassDeclaration": false,
          "ClassExpression": false,
          "FunctionDeclaration": true,
          "FunctionExpression": false,
          "MethodDefinition": false
        }
      }
    ],
    "jsdoc/check-tag-names": [
      "off"
    ],
    "jsdoc/no-undefined-types": [
      "off"
    ],
    "jsdoc/valid-types": [
      "off"
    ],
    "jsdoc/check-param-names": [
      "off",
      {
        "checkDestructured": false
      }
    ],
    "jsdoc/check-values": [
      "off"
    ],
    "jsdoc/tag-lines": [
      "off",
      "any",
      {
        "startLines": 1
      }
    ],
    "jsdoc/check-access": [
      "warn"
    ],
    "jsdoc/check-alignment": [
      "warn"
    ],
    "jsdoc/check-examples": [
      "off"
    ],
    "jsdoc/check-indentation": [
      "off"
    ],
    "jsdoc/check-line-alignment": [
      "off"
    ],
    "jsdoc/check-property-names": [
      "warn"
    ],
    "jsdoc/check-syntax": [
      "off"
    ],
    "jsdoc/check-types": [
      "warn"
    ],
    "jsdoc/empty-tags": [
      "warn"
    ],
    "jsdoc/implements-on-classes": [
      "warn"
    ],
    "jsdoc/informative-docs": [
      "off"
    ],
    "jsdoc/match-description": [
      "off"
    ],
    "jsdoc/match-name": [
      "off"
    ],
    "jsdoc/multiline-blocks": [
      "warn"
    ],
    "jsdoc/no-bad-blocks": [
      "off"
    ],
    "jsdoc/no-blank-block-descriptions": [
      "off"
    ],
    "jsdoc/no-defaults": [
      "off"
    ],
    "jsdoc/no-missing-syntax": [
      "off"
    ],
    "jsdoc/no-multi-asterisks": [
      "warn"
    ],
    "jsdoc/no-restricted-syntax": [
      "off"
    ],
    "jsdoc/no-types": [
      "off"
    ],
    "jsdoc/require-asterisk-prefix": [
      "off"
    ],
    "jsdoc/require-description": [
      "off"
    ],
    "jsdoc/require-description-complete-sentence": [
      "off"
    ],
    "jsdoc/require-example": [
      "off"
    ],
    "jsdoc/require-file-overview": [
      "off"
    ],
    "jsdoc/require-hyphen-before-param-description": [
      "off"
    ],
    "jsdoc/require-param-name": [
      "warn"
    ],
    "jsdoc/require-property": [
      "warn"
    ],
    "jsdoc/require-property-description": [
      "warn"
    ],
    "jsdoc/require-property-name": [
      "warn"
    ],
    "jsdoc/require-property-type": [
      "warn"
    ],
    "jsdoc/require-returns-check": [
      "warn"
    ],
    "jsdoc/require-returns-description": [
      "warn"
    ],
    "jsdoc/require-throws": [
      "off"
    ],
    "jsdoc/require-yields": [
      "warn"
    ],
    "jsdoc/require-yields-check": [
      "warn"
    ],
    "jsdoc/sort-tags": [
      "off"
    ],
    "jsdoc/text-escaping": [
      "off"
    ],
    "@typescript-eslint/adjacent-overload-signatures": [
      "error"
    ],
    "@typescript-eslint/ban-ts-comment": [
      "error"
    ],
    "no-array-constructor": [
      "off"
    ],
    "@typescript-eslint/no-array-constructor": [
      "error"
    ],
    "no-empty-function": [
      "off"
    ],
    "@typescript-eslint/no-extra-non-null-assertion": [
      "error"
    ],
    "no-extra-semi": [
      "off"
    ],
    "@typescript-eslint/no-extra-semi": [
      "error"
    ],
    "@typescript-eslint/no-inferrable-types": [
      "error"
    ],
    "no-loss-of-precision": [
      "off"
    ],
    "@typescript-eslint/no-loss-of-precision": [
      "error"
    ],
    "@typescript-eslint/no-misused-new": [
      "error"
    ],
    "@typescript-eslint/no-non-null-asserted-optional-chain": [
      "error"
    ],
    "@typescript-eslint/no-this-alias": [
      "error"
    ],
    "@typescript-eslint/no-unnecessary-type-constraint": [
      "error"
    ],
    "@typescript-eslint/no-var-requires": [
      "error"
    ],
    "@typescript-eslint/prefer-as-const": [
      "error"
    ],
    "@typescript-eslint/prefer-namespace-keyword": [
      "error"
    ],
    "@typescript-eslint/triple-slash-reference": [
      "error"
    ],
    "constructor-super": [
      "off"
    ],
    "getter-return": [
      "off"
    ],
    "no-const-assign": [
      "off"
    ],
    "no-dupe-args": [
      "off"
    ],
    "no-dupe-class-members": [
      "off"
    ],
    "no-dupe-keys": [
      "off"
    ],
    "no-func-assign": [
      "off"
    ],
    "no-import-assign": [
      "off"
    ],
    "no-new-symbol": [
      "off"
    ],
    "no-obj-calls": [
      "off"
    ],
    "no-redeclare": [
      "off"
    ],
    "no-setter-return": [
      "off"
    ],
    "no-this-before-super": [
      "off"
    ],
    "no-undef": [
      "off"
    ],
    "no-unreachable": [
      "off"
    ],
    "no-unsafe-negation": [
      "off"
    ],
    "no-var": [
      "error"
    ],
    "prefer-const": [
      "error",
      {
        "destructuring": "all",
        "ignoreReadBeforeAssign": false
      }
    ],
    "prefer-rest-params": [
      "error"
    ],
    "prefer-spread": [
      "error"
    ],
    "valid-typeof": [
      "off"
    ],
    "curly": [
      "error",
      "multi-line"
    ],
    "eol-last": [
      "error"
    ],
    "keyword-spacing": [
      "error"
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "max-len": [
      "error",
      {
        "code": 100,
        "tabWidth": 4,
        "ignoreStrings": true,
        "ignoreRegExpLiterals": true,
        "ignoreUrls": true
      }
    ],
    "no-trailing-spaces": [
      "error",
      {
        "skipBlankLines": true,
        "ignoreComments": false
      }
    ],
    "no-self-assign": [
      "off"
    ],
    "quotes": [
      "off",
      "single",
      {
        "allowTemplateLiterals": true
      }
    ],
    "semi": [
      "error",
      "always"
    ],
    "space-before-function-paren": [
      "error",
      "never"
    ],
    "space-infix-ops": [
      "error"
    ],
    "strict": [
      "off"
    ],
    "no-console": [
      "off"
    ],
    "no-multi-spaces": [
      "error",
      {
        "ignoreEOLComments": true
      }
    ],
    "operator-linebreak": [
      "error",
      "after"
    ],
    "spaced-comment": [
      "error",
      "always",
      {
        "block": {
          "balanced": true
        },
        "exceptions": [
          "*"
        ]
      }
    ],
    "camelcase": [
      "error",
      {
        "properties": "never",
        "allow": [
          "^opt_",
          "^_opt_",
          "^testOnly_"
        ],
        "ignoreDestructuring": false,
        "ignoreImports": false,
        "ignoreGlobals": false
      }
    ],
    "indent": [
      "off",
      2,
      {
        "CallExpression": {
          "arguments": 2
        },
        "FunctionDeclaration": {
          "body": 1,
          "parameters": 2
        },
        "FunctionExpression": {
          "body": 1,
          "parameters": 2
        },
        "MemberExpression": 2,
        "ObjectExpression": 1,
        "SwitchCase": 1,
        "ignoredNodes": [
          "ConditionalExpression"
        ],
        "flatTernaryExpressions": false,
        "offsetTernaryExpressions": false,
        "ignoreComments": false
      }
    ],
    "new-cap": [
      "off"
    ],
    "brace-style": [
      "error",
      "1tbs",
      {
        "allowSingleLine": true
      }
    ],
    "guard-for-in": [
      "off"
    ],
    "comma-dangle": [
      "error",
      {
        "arrays": "always-multiline",
        "objects": "always-multiline",
        "imports": "always-multiline",
        "exports": "always-multiline",
        "functions": "ignore"
      }
    ],
    "no-cond-assign": [
      0
    ],
    "no-irregular-whitespace": [
      2
    ],
    "no-unexpected-multiline": [
      2
    ],
    "no-caller": [
      2
    ],
    "no-extend-native": [
      2
    ],
    "no-extra-bind": [
      2
    ],
    "no-multi-str": [
      2
    ],
    "no-new-wrappers": [
      2
    ],
    "no-throw-literal": [
      2
    ],
    "no-with": [
      2
    ],
    "prefer-promise-reject-errors": [
      2
    ],
    "array-bracket-newline": [
      0
    ],
    "array-bracket-spacing": [
      2,
      "never"
    ],
    "array-element-newline": [
      0
    ],
    "block-spacing": [
      2,
      "never"
    ],
    "comma-spacing": [
      2
    ],
    "comma-style": [
      2
    ],
    "computed-property-spacing": [
      2
    ],
    "key-spacing": [
      2
    ],
    "no-mixed-spaces-and-tabs": [
      2
    ],
    "no-multiple-empty-lines": [
      2,
      {
        "max": 2
      }
    ],
    "no-new-object": [
      2
    ],
    "no-tabs": [
      2
    ],
    "object-curly-spacing": [
      2
    ],
    "one-var": [
      2,
      {
        "var": "never",
        "let": "never",
        "const": "never"
      }
    ],
    "padded-blocks": [
      2,
      "never"
    ],
    "quote-props": [
      2,
      "consistent"
    ],
    "semi-spacing": [
      2
    ],
    "space-before-blocks": [
      2
    ],
    "switch-colon-spacing": [
      2
    ],
    "arrow-parens": [
      2,
      "always"
    ],
    "generator-star-spacing": [
      2,
      "after"
    ],
    "rest-spread-spacing": [
      2
    ],
    "yield-star-spacing": [
      2,
      "after"
    ],
    "for-direction": [
      "error"
    ],
    "no-async-promise-executor": [
      "error"
    ],
    "no-case-declarations": [
      "error"
    ],
    "no-class-assign": [
      "error"
    ],
    "no-compare-neg-zero": [
      "error"
    ],
    "no-constant-condition": [
      "error"
    ],
    "no-control-regex": [
      "error"
    ],
    "no-debugger": [
      "error"
    ],
    "no-delete-var": [
      "error"
    ],
    "no-dupe-else-if": [
      "error"
    ],
    "no-duplicate-case": [
      "error"
    ],
    "no-empty": [
      "error"
    ],
    "no-empty-character-class": [
      "error"
    ],
    "no-empty-pattern": [
      "error"
    ],
    "no-ex-assign": [
      "error"
    ],
    "no-extra-boolean-cast": [
      "error"
    ],
    "no-fallthrough": [
      "error"
    ],
    "no-global-assign": [
      "error"
    ],
    "no-inner-declarations": [
      "error"
    ],
    "no-invalid-regexp": [
      "error"
    ],
    "no-misleading-character-class": [
      "error"
    ],
    "no-nonoctal-decimal-escape": [
      "error"
    ],
    "no-octal": [
      "error"
    ],
    "no-prototype-builtins": [
      "error"
    ],
    "no-regex-spaces": [
      "error"
    ],
    "no-shadow-restricted-names": [
      "error"
    ],
    "no-sparse-arrays": [
      "error"
    ],
    "no-unsafe-finally": [
      "error"
    ],
    "no-unsafe-optional-chaining": [
      "error"
    ],
    "no-unused-labels": [
      "error"
    ],
    "no-useless-backreference": [
      "error"
    ],
    "no-useless-catch": [
      "error"
    ],
    "no-useless-escape": [
      "error"
    ],
    "require-yield": [
      "error"
    ],
    "use-isnan": [
      "error"
    ]
  },
  "settings": {
    "jsdoc": {
      "mode": "typescript"
    }
  }
}
```

## Environment

- Node version: v16.18.0 and v20.0.0 at least.
- ESLint version: v8.37.0
- `eslint-plugin-jsdoc` version: 43.0.6

**Comment by cpcallen** [comment]
Gosh you are fast!  I can confirm that this indeed does fix the problem we were running in to.  Thank you very much.
